### ISSUE TITLE
Missing Prometheus annotation inside Network Costs template

### ISSUE THREAD

**Issue by pierluigilenoci** [kubecost/cost-analyzer-helm-chart#770]
**Describe the bug**

I open this as a result of the tests done on the new [release](https://github.com/kubecost/cost-analyzer-helm-chart/releases/tag/v1.74.0). 

Prometheus does not scrap the Network Costs pod.

Ref: https://github.com/kubecost/cost-analyzer-helm-chart/issues/741#issuecomment-777627784 

**Proposed solution**

Here:
https://github.com/kubecost/cost-analyzer-helm-chart/blob/47af25f922a6bf0c329f25137f5f8817b2cc91d6/cost-analyzer/templates/cost-analyzer-network-costs-template.yaml#L16-L18
is missing this:
```yaml
annotations:
  prometheus.io/scrape: 'true'
```

Manually added to the Daemonset everything started working perfectly.

**Comment by dwbrown2** [comment]
@pierluigilenoci thanks for the great report and suggestion! most teams directly add this scrape when using BYO Prom instead of using service discovery. I definitely see your point though.

@mbolt35 thoughts? seems like a very easy addition, but would this lead to double scraping with our bundled Prom? any other unintended consequences?

**Comment by mbolt35** [comment]
> @mbolt35 thoughts? seems like a very easy addition, but would this lead to double scraping with our bundled Prom? any other unintended consequences?

Yeah, does seem specific to bring your own prom, since our helm chart is configured for the network costs scrape. Does prometheus not recognize the same source? I'll check this out.

**Comment by dwbrown2** [comment]
@pierluigilenoci here's how you can do this today: https://github.com/kubecost/cost-analyzer-helm-chart/blob/6c0975614b4a6854be602d1a6f9506ce8b80abdc/cost-analyzer/values.yaml#L294-L301

@mbolt35 if we do this via service discovery, then I believe we would want to remove the config above.

**Comment by pierluigilenoci** [comment]
@dwbrown2 I prefer a more general solution lowering the impact of external tool configuration changes.
IMHO, a tool should be as autonomous as possible in its configuration, minimizing the impact on the ecosystem in which it is integrated.

**Comment by pierluigilenoci** [comment]
@dwbrown2 however, the solution you are using now is not documented on the [page](https://github.com/kubecost/docs/blob/master/custom-prom.md) for the use of custom Prometheus.

**Comment by mbolt35** [comment]
@pierluigilenoci With the network-costs pod, I just want to make sure the requirements are understood. Most deployment pods are scraped as a single service scrape, but we want all the daemonset pods to be scraped individually. 

I have some concerns due to the "multiple scrape" issue that @dwbrown2 mentioned, whereas we have the issue where both the autodiscovery and helm configurations both hold. Looking in our chart, I only see this annotation in `node-exporter-service` and thanos has a configuration option to allow it: 

```
{{- if .Values.store.metrics.annotations.enabled  }}
      prometheus.io/scrape: "true"
      prometheus.io/port: "{{ .Values.store.http.port }}"
      {{- end }}
```

Does something like the above seem adequate?

**Comment by pierluigilenoci** [comment]
@mbolt35 yep seems adequate.

**Comment by mbolt35** [comment]
@pierluigilenoci Thanks for the feedback here! PR is here: https://github.com/kubecost/cost-analyzer-helm-chart/pull/795
Should be part of next release.

**Comment by pierluigilenoci** [comment]
@mbolt35 when will this PR be released? Because it is not inside the v1.76.0 like you said.

**Comment by dwbrown2** [comment]
Merged now, will be available in next release!
